test/trusted_resources_test.go

```diff
@@ -0,0 +1,315 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"github.com/tektoncd/pipeline/test/parse"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/system"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+var (
+	neededFeatureFlags = map[string]string{
+		"resource-verification-mode": "enforce",
```

Comment:
if i'm understanding how `requireAnyGate` works correctly, I think there might be something odd about including this flag here? it seems like `requireAnyGate` will run the current test if any of the passed in flags are set - since this test is providing both `resource-verification-mode` and `enable-api-fields`, am I right that that means the test skipping/running would be:

1. `resource-verification-mode` set, `enable-api-fields`, set <-- run the test
2. `resource-verification-mode` unset (which is what we'd expect i think), `enable-api-fields`, set <-- run the test
3.`resource-verification-mode` set, `enable-api-fields`, unset <-- run the test
4.`resource-verification-mode` unset, `enable-api-fields`, unset <-- skip the test

i.e. it would only skip the test in case (4)? i think we'd want to skip the test in case (3) as well? long story short I'm wondering if we should _only_ pass `enable-api-fields` to `requireAnyGate`

Reply:
Yes, you're totally correct! It will run if any of these 2 are set. It totally makes sense that we only pass `enable-api-fields`, I will change it accordingly.